lib/repositories/interface/i_movies_repository.dart

```diff
@@ -0,0 +1,7 @@
+import '../../data_models/movie.dart';
+
+abstract class IMoviesRepository {
+  Future<List<Movie>> fetchMovies();
+
+  Future<List<Movie>> parseMovies();
```

Comment:
I wouldn't add this `parseMovies` method as an interface method. You only need a `fetch` method, the implementation will take care of the **where** you get the data.
Same for the genres repo